test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partnership/PartnershipPartnersPhoneNumberControllerSpec.scala

```diff
@@ -0,0 +1,140 @@
+
+class PartnershipPartnersPhoneNumberControllerSpec extends ControllerSpec with DefaultAwaitTimeout {
+
+  private val page = mock[phone_number_page]
+  private val mcc  = stubMessagesControllerComponents()
+
+  private val controller =
+    new PartnershipOtherPartnerPhoneNumberController(authenticate = mockAuthAction,
+                                                     journeyAction = mockJourneyAction,
+                                                     registrationConnector =
+                                                       mockRegistrationConnector,
+                                                     mcc = mcc,
+                                                     page = page
+    )
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    when(page.apply(any(), any(), any(), any())(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit = {
+    reset(page)
+    super.afterEach()
+  }
+
+  def registrationWithInflightOtherPartnerJourney =
+    aRegistration(withPartnershipDetails(Some(generalPartnershipDetails))).withInflightPartner(
+      Some(
+        Partner(contactDetails =
+                  Some(PartnerContactDetails(firstName = Some("John"), lastName = Some("Smith"))),
+                partnerType = Some(PartnerTypeEnum.SOLE_TRADER)
+        )
+      )
+    )
+
+  def registrationWithInflightOtherPartnerJourneyAndPhoneNumber =
+    aRegistration(withPartnershipDetails(Some(generalPartnershipDetails))).withInflightPartner(
+      Some(
+        Partner(
+          contactDetails = Some(
+            PartnerContactDetails(firstName = Some("John"),
+                                  lastName = Some("Smith"),
+                                  phoneNumber = Some("12345678")
+            )
+          ),
+          partnerType = Some(PartnerTypeEnum.SOLE_TRADER)
+        )
+      )
+    )
+
+  "PartnershipOtherPartnerPhoneNumberController" should {
+
+    "return 200" when {
+      "user is authorised, a registration already exists with already collected contact name and display page method is invoked" in {
+        authorizedUser()
+        mockRegistrationFind(registrationWithInflightOtherPartnerJourney)
+
+        val result = controller.displayPage()(getRequest())
+
+        status(result) mustBe OK
+      }
+
+      "user is authorised, a registration already exists with already collected contact name and phone number display page method is invoked" in {
+        authorizedUser()
+        mockRegistrationFind(registrationWithInflightOtherPartnerJourneyAndPhoneNumber)
+
+        val result = controller.displayPage()(getRequest())
+
+        status(result) mustBe OK
+      }
+    }
+
+    "return 404" when {
+      "user is authorised but does not have an inflight journey and display page method is invoked" in {
+        authorizedUser()
+
+        val result = controller.displayPage()(getRequest())
+
+        status(result) mustBe NOT_FOUND
+      }
+    }
+
```

Comment:
I think we should have a test to verify that the registration is updated as expected when form input is valid and that routing is as expected.